### DISCUSSION TITLE
I created collection but when i create collection inside fields i got this issues

### DISCUSSION THREAD

**Discussion by raselmiazi** [directus/directus#22331] [Feature Requests]
### Summary

I got the error message
![image](https://github.com/directus/directus/assets/163228495/0deda9d3-e1e1-4e3b-b86e-acf35eb22ee5)

here is my All code 
import {
    createDirectus,
    rest,
    withToken,
    createCollection,
    createField
} from '@directus/sdk';

const client = createDirectus(process.env.DIRECTUS_HOST).with(rest());

// Collection schemas
const collectionSchemas = {
    "about_us": {
        meta: {
            collection: 'about_us',
            icon: 'supervised_user_circle',
            note: null,
            display_template: null,
            hidden: false,
            singleton: false,
            translations: null,
            archive_field: null,
            archive_app_filter: true,
            archive_value: null,
            unarchive_value: null,
            sort_field: null,
            accountability: 'all',
            color: null,
            item_duplication_fields: null,
            sort: null,
            group: null,
            collapse: 'open',
            preview_url: null,
            versioning: false
        },
        schema: {
            schema: 'public',
            name: 'about_us',
            comment: null,
            fields: {
                "id": "integer NOT NULL",
                "status": "character varying(255) DEFAULT 'draft'::character varying NOT NULL",
                "user_created": "uuid",
                "date_created": "timestamp with time zone",
                "user_updated": "uuid",
                "date_updated": "timestamp with time zone",
                "Page_name": "character varying(255)",
                "Browser_title": "character varying(255)",
                "Disclaimer": "boolean",
                "Sitemap": "boolean",
                "Page_URL": "character varying(255)",
                "Meta_description": "character varying(255)",
                "Keywords": "character varying(255)"
            }
        }
    },
    "ESG_info_page": {
        meta: {
            collection: 'ESG_info_page',
            icon: 'supervised_user_circle',
            note: null,
            display_template: null,
            hidden: false,
            singleton: false,
            translations: null,
            archive_field: null,
            archive_app_filter: true,
            archive_value: null,
            unarchive_value: null,
            sort_field: null,
            accountability: 'all',
            color: null,
            item_duplication_fields: null,
            sort: null,
            group: null,
            collapse: 'open',
            preview_url: null,
            versioning: false
        },
        schema: {
            schema: 'public',
            name: 'ESG_info_page',
            comment: null,
            fields: {
                "id": "integer NOT NULL",
                "status": "character varying(255) DEFAULT 'draft'::character varying NOT NULL",
                "user_created": "uuid",
                "date_created": "timestamp with time zone",
                "user_updated": "uuid",
                "date_updated": "timestamp with time zone",
                "Page_name": "character varying(255)",
                "Browser_title": "character varying(255)",
                "Disclaimer": "boolean",
                "Sitemap": "boolean",
                "Page_URL": "character varying(255)",
                "Meta_description": "character varying(255)",
                "Keywords": "character varying(255)"
            }
        }
    },
    "AMC_info_page": {
        meta: {
            collection: 'AMC_info_page',
            icon: 'supervised_user_circle',
            note: null,
            display_template: null,
            hidden: false,
            singleton: false,
            translations: null,
            archive_field: null,
            archive_app_filter: true,
            archive_value: null,
            unarchive_value: null,
            sort_field: null,
            accountability: 'all',
            color: null,
            item_duplication_fields: null,
            sort: null,
            group: null,
            collapse: 'open',
            preview_url: null,
            versioning: false
        },
        schema: {
            schema: 'public',
            name: 'AMC_info_page',
            comment: null,
            fields: {
                "id": "integer NOT NULL",
                "status": "character varying(255) DEFAULT 'draft'::character varying NOT NULL",
                "user_created": "uuid",
                "date_created": "timestamp with time zone",
                "user_updated": "uuid",
                "date_updated": "timestamp with time zone",
                "Page_name": "character varying(255)",
                "Browser_title": "character varying(255)",
                "Disclaimer": "boolean",
                "Sitemap": "boolean",
                "Page_URL": "character varying(255)",
                "Meta_description": "character varying(255)",
                "Keywords": "character varying(255)"
            }
        }
    }
};

(async () => {
    try {
        for (const collectionName in collectionSchemas) {
            console.log(`Creating fields for collection ${collectionName}...`);
           
            // Extract fields for the collection
            const { schema } = collectionSchemas[collectionName];
            const fields = collectionSchemas[collectionName].schema.fields;
            console.log(fields);
            console.log(schema);
            await client.request(
                withToken(
                    process.env.ADMIN_TOKEN,
                    createCollection({
                        collection: collectionName,
                        schema: {
                            fields: schema.fields
                        },
                        meta: {
                            schema: schema
                        }
                    })
                )
            )

            // Iterate over each field and create it
            for (const fieldName in fields) {
                const fieldType = fields[fieldName];
                console.log(fieldName)

                // Create the field with the admin token
                await client.request(
                    withToken(
                        process.env.ADMIN_TOKEN,
                        createField({
                            field: fieldName,
                            type: fieldType,
                            // field_field: value,
                        })
                    )
                );

                // console.log(`Field ${fieldName} created successfully.`);
            }

            console.log(`All fields for collection ${collectionName} created successfully.`);
        }
    } catch (error) {
        console.error('Error creating fields:', error);
    }
})();

console.log(`[IMPORT DONE]`);

### Motivation

![image](https://github.com/directus/directus/assets/163228495/b42dba82-946c-459e-85fd-f90c2e949d17)

### Detailed Design

![image](https://github.com/directus/directus/assets/163228495/41ee7e86-0b14-4563-b0f7-6eaed8e2b4be)

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

![image](https://github.com/directus/directus/assets/163228495/6a8078d1-ecb0-4973-8fa8-ab6f3ea3d0de)

### Alternatives

![image](https://github.com/directus/directus/assets/163228495/fb44ad58-cb80-42d7-a39f-731793ec1b72)

### Adoption Strategy

![image](https://github.com/directus/directus/assets/163228495/7ca6b2ce-9499-41d7-96ae-bd0c48687eab)

**Comment by br41nslug** [comment]
This is not a feature request, judging from the error you are likely building an invalid payload somewhere. Feel free to ask the community for support in [Discord](https://directus.chat/)
